extensions/azdata/src/azdata.ts

```diff
@@ -90,10 +102,19 @@ class AzdataTool implements IAzdataTool {
 		return this.executeCommand<void>(['login', '-e', endpoint, '-u', username], { 'AZDATA_PASSWORD': password });
 	}
 
+	/**
+	 * Gets the output of running '--version' command on the azdata tool.
+	 * It also updates the cachedVersion property based on the return value from the tool.
+	 */
 	public async version(): Promise<azdataExt.AzdataOutput<string>> {
-		const output = await this.executeCommand<string>(['--version']);
-		this.toolVersion = parseVersion(output.stdout[0]);
-		return output;
+		const output = await executeCommand(`"${this.path}"`, ['--version']);
+		this.cachedVersion = new SemVer(parseVersion(output.stdout[0]));
```

Comment:
output.stdout is a single string so indexing into it doesn't seem correct.

Reply:
done